src/libtiled/maprenderer.cpp

```diff
@@ -335,7 +350,17 @@ void CellRenderer::render(const Cell &cell, const QPointF &pos, const QSizeF &si
     const QRectF source(0, 0, fragment.width, fragment.height);
 
     mPainter->setTransform(transform);
-    mPainter->drawPixmap(target, image, source);
+    QPixmap tintedimg = image;
+    if(layer){
```

Comment:
When is layer not a valid pointer? Could it be turned into a reference?

Reply:
Now I'm storing the tintcolor in the cellrenderer as you said, so passing around a pointer or reference to the layer becomes unnecessary.